bin/check-dns-zone.rb

```diff
@@ -0,0 +1,204 @@
+#! /usr/bin/env ruby
+#
+#   check-dns
+#
+# DESCRIPTION:
+#   This plugin checks DNS Zone using dnsruby and whois.
+#
+# OUTPUT:
+#   plain text
+#
+# PLATFORMS:
+#   Linux, BSD
+#
+# DEPENDENCIES:
+#   gem: sensu-plugin
+#   gem: dnsruby
+#   gem: whois
+#
+# USAGE:
+#   example commands
+#
+# NOTES:
+#   Does it behave differently on specific platforms, specific use cases, etc
+#
+# LICENSE:
+#   Zubov Yuri <yury.zubau@gmail.com> sponsored by Actility, https://www.actility.com
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'dnsruby'
+require 'whois'
+require 'whois/parser'
+require 'ipaddr'
+require 'resolv'
+
+#
+# DNS
+#
+class DNSZone < Sensu::Plugin::Check::CLI
+  option :domain,
+         description: 'Domain to resolve (or ip if type PTR)',
+         short: '-d DOMAIN',
+         long: '--domain DOMAIN'
+
+  option :threshold,
+         description: 'Percentage of DNS queries that must succeed',
+         short: '-l PERCENT',
+         long: '--threshold PERCENT',
+         proc: proc(&:to_i),
+         default: 100
+
+  option :timeout,
+         description: 'Set timeout for query',
+         short: '-T TIMEOUT',
+         long: '--timeout TIMEOUT',
+         proc: proc(&:to_i),
+         default: 5
+
+  option :warn_only,
+         description: 'Warn instead of critical on failure',
+         short: '-w',
+         long: '--warn-only',
+         boolean: true
+
+  def resolve_ns
+    Resolv::DNS.new.getresources(config[:domain], Resolv::DNS::Resource::IN::NS).map { |e| e.name.to_s }
+  end
+
+  def check_nss(entries)
```

Comment:
Can we use a better name? `check_nss` made me think for a minute you were talking about mozilla's [Network Security Services Library](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/NSS)